plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/OpenShiftLaunchController.java

```diff
@@ -50,20 +67,83 @@ protected static IControllableServerBehavior getServerBehavior(ILaunchConfigurat
 	@Override
 	public void launch(ILaunchConfiguration configuration, String mode,
 			ILaunch launch, IProgressMonitor monitor) throws CoreException {
-		IControllableServerBehavior beh = getServerBehavior(configuration);
-		if( beh != null ) {
-			((ControllableServerBehavior)beh).setServerStarting();
-			int state = pollState();
-			if( state == IServer.STATE_STARTED) {
-				((ControllableServerBehavior)beh).setServerStarted();
-				((ControllableServerBehavior)getControllableBehavior()).setRunMode(mode);
-			} else {
-				((ControllableServerBehavior)beh).setServerStopped();
-				((ControllableServerBehavior)getControllableBehavior()).setRunMode(null);
-			}
+		IControllableServerBehavior serverBehavior = getServerBehavior(configuration);
+		if( !(serverBehavior instanceof ControllableServerBehavior )) {
+			throw toCoreException("Unable to find a IControllableServerBehavior instance");
+		}
+		ControllableServerBehavior beh = (ControllableServerBehavior) serverBehavior;
+		
+		beh.setServerStarting();
+		
+		IServer server = beh.getServer();
+		
+		IDeploymentConfig dc = OpenShiftServerUtils.getDeploymentConfig(server);
+		DebuggingContext debugContext = OpenShiftDebugUtils.get().getDebuggingContext(dc);
```

Comment:
However, `OpenShiftDebugUtils.get().getDebuggingContext(IDeploymentConfig)` can return `null`, which means that a `null` `debugContext` will be passed to the `startDebugging` and `stopDebugging` methods, which will cause NPEs